scripts/modules/data/models/behaviors.mjs

```diff
@@ -0,0 +1,119 @@
+import {MODULE} from "../../constants.mjs";
+
+const {BooleanField, HTMLField, NumberField, SchemaField, StringField} = foundry.data.fields;
+
+/**
+ * Initialization function to register data models.
+ */
+export default function initializeBehaviors() {
+  CONFIG.RegionBehavior.dataModels[`${MODULE.ID}.trap`] = TrapData;
+  CONFIG.RegionBehavior.typeIcons[`${MODULE.ID}.trap`] = "fa-solid fa-person-falling-burst";
+}
+
+/**
+ * Behavior type that configures a trap that prompts a saving throw and deals damage.
+ */
+class TrapData extends foundry.data.regionBehaviors.RegionBehaviorType {
+  /** @override */
+  static events = {
+    [CONST.REGION_EVENTS.TOKEN_ENTER]: foundry.utils.debounce(TrapData.#onTokenMoveIn, 150)
+  };
+
+  /* -------------------------------------------------- */
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["MYTHACRI.REGION.TRAP"];
+
+  /* -------------------------------------------------- */
+
+  /** @override */
+  static defineSchema() {
+    return {
+      save: new SchemaField({
+        targetValue: new NumberField({integer: true, initial: 15}),
+        type: new StringField({
+          choices: CONFIG.DND5E.abilities, required: true, initial: "dex"
+        })
+      }),
+      damage: new SchemaField({
+        formula: new dnd5e.dataModels.fields.FormulaField({
+          required: true, validate: value => Roll.validate(value)
+        }),
+        type: new StringField({
+          choices: CONFIG.DND5E.damageTypes, required: true, initial: "piercing"
+        })
+      }),
+      config: new SchemaField({
+        toggle: new BooleanField({initial: true}),
+        message: new HTMLField({required: true})
+      })
+    };
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * The script that is executed when a token moves into the region.
+   * @param {RegionEvent} event     The region event data from triggering this behavior.
+   */
+  static async #onTokenMoveIn(event) {
+    const {data: {token}, user} = event;
+
+    const gm = game.users.activeGM;
+    if (![gm, user].includes(game.user)) return;
+
+    const actor = token.actor;
+    if (!actor) {
+      ui.notifications.error("MYTHACRI.REGION.TRAP.WARNINGS.ActorMissing", {localize: true});
+      return;
+    }
+
+    if (!this.damage.formula) {
+      ui.notifications.error("MYTHACRI.REGION.TRAP.WARNINGS.FormulaMissing", {localize: true});
+      return;
+    }
+
+    // GM only functions.
+    if (game.user === gm) {
+      if (this.config.toggle) await this.behavior.update({disabled: true});
```

Comment:
If this behavior should be performed once only, this gets allocated to the GM to disable it.